### ISSUE TITLE
Can't install Zip to Jibri Docker

### ISSUE THREAD

**Issue by Drumsk8** [jitsi/docker-jitsi-meet#1406]
Hello, 

Firstly thank you for a wonderful platform!

I am currently in need of Zip so that I can password protect some videos after Jibri has created them. I've tried adding a command to the docker compose file like this:

  image: jitsi/jibri:${JITSI_IMAGE_VERSION:-stable-7648-4}
  command: "apt update && apt install -y zip"

This command works if I login to the container and run it, but if I add it to the compose file like above, it breaks the framebuffer and causes jibri to fail every time and enter a boot loop. 

Please could zip be added to the base container as tar can't be used for this purpose unfortunately. If it can't be added to the base image, please can someone help me with getting a working command with which works in a compose file.

Thank you in advance.

**Comment by saghul** [comment]
You could extend the Jibri image by creating your own and just adding zip to it. It would be a RUN line in your Dockerfile.

**Comment by Drumsk8** [comment]
Thank you for getting back to me. 

I've copied the dockerfile from[ latest release](https://github.com/jitsi/docker-jitsi-meet/releases/tag/stable-7648-4) and added zip to the end of the RUN command as you suggested. That appears to be successful from the logs.

However, unfortunately the build fails for a different reason, permissions on /usr/bin/install-chrome.sh which follows after the main package installations. 

------
#0 67.64 /bin/sh: 1: /usr/bin/install-chrome.sh: Permission denied
------
failed to solve: executor failed running [/bin/sh -c apt-dpkg-wrap apt-get update &&     apt-dpkg-wrap apt-get install -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" jibri libgl1-mesa-dri procps jitsi-upload-integrations jq pulseaudio dbus dbus-x11 rtkit &&     /usr/bin/install-chrome.sh &&     apt-cleanup &&     adduser jibri rtkit]: exit code: 126

I am assuming that /usr/bin/install-chrome.sh is within the container so, either the buildfile from the latest release is incorrect some how or I am missing something. 

Sorry to be a pain.

**Comment by saghul** [comment]
Are you running Docker build as root by any chance?

**Comment by Drumsk8** [comment]
That's correct.

**Comment by Drumsk8** [comment]
It dawned on me that the image is using root to build itself, so possible security escalation point there.

I see that there's a rootless mode for the docker engine now, been awhile since I'd dealt with docker manually like this so will be investigating that now. No doubt from your question, I'll find my answer in a non-root execution of the build command. 

Thank you for your assistance.

**Comment by saghul** [comment]
Adding your user to the "docker" group should do it. Then built it without root and you should be good.

**Comment by Drumsk8** [comment]
Hi, 

Sorry, I had to get some sleep. I've just tried adding my user to docker group and I get the same error on build. 

I have no problem using this user to run the compose if not building from dockerfile. So it's group is applied correctly although I've made this user apart of wheel and sudoers as well, I wouldn't of thought that would be an issue.

I am trying my command as:
$ docker-compose up -d --build --force-recreate

Here's the jibri part of my compose file:

This is my config mapping from .env: CONFIG=/opt/docker/jitsi-tsw/jitsi-meet-cfg

   # Video Recorder
    jibri-tsw:
        container_name: jitsi-jibri-tsw
        # image: jitsi/jibri:${JITSI_IMAGE_VERSION:-stable-7648-4}
        # command: "apt update && apt install -y zip"
        build: jibri/.
        restart: ${RESTART_POLICY:-unless-stopped}
        volumes:
            - ${CONFIG}/jibri:/config:Z
            - /var/www/vhosts/domain/html/group-chat:/home/jibri/recordings/group_chat:rw
            - /var/www/vhosts/domain/html/live-archive:/home/jibri/recordings/live_stream:rw
            - /var/www/vhosts/domain/html/1-2-1:/home/jibri/recordings/1-2-1:rw
        shm_size: '4gb' 
        cap_add:
            - SYS_ADMIN
        environment:
            - CHROMIUM_FLAGS
            - DISPLAY=:0
            - ENABLE_STATS_D
            - JIBRI_HTTP_API_EXTERNAL_PORT
            - JIBRI_HTTP_API_INTERNAL_PORT
            - JIBRI_RECORDING_RESOLUTION
            - JIBRI_USAGE_TIMEOUT
            - JIBRI_XMPP_USER
            - JIBRI_XMPP_PASSWORD
            - JIBRI_BREWERY_MUC
            - JIBRI_RECORDER_USER
            - JIBRI_RECORDER_PASSWORD
            - JIBRI_RECORDING_DIR
            - JIBRI_FINALIZE_RECORDING_SCRIPT_PATH
            - JIBRI_STRIP_DOMAIN_JID
            - PUBLIC_URL
            - TZ
            - XMPP_AUTH_DOMAIN
            - XMPP_DOMAIN
            - XMPP_INTERNAL_MUC_DOMAIN
            - XMPP_MUC_DOMAIN
            - XMPP_RECORDER_DOMAIN
            - XMPP_SERVER
            - XMPP_PORT
            - XMPP_TRUST_ALL_CERTS
        depends_on:
            - jicofo-tsw
        networks:
            meet.jitsi-tsw:

Docker file is exactly as is from the release. I have no problems with jibri if I don't try using the dockerfile.

System is:
Kernel: 4.18.0-372.13.1.el8_6.x86_64
OS: Rocky Linux release 8.6 (Green Obsidian)
Docker Engine: 20.10.17, build 100c701
Docker Compose: v2.9.0

Is there anything here you can see I am doing wrong?

Thank you for your assistance.

**Comment by saghul** [comment]
Maybe try this instead: go into the jibri directory and edit the Dockerfile to add zip.

Go to the root directory and run make build_Jibri that will build the libro image.

Now modify the compose file to use the "latest" tag for Jibri.

Then try to start the compose setup normally.

**Comment by Drumsk8** [comment]
I'd made a poor judgment and assumed the files for the container was being streamed in, I overlooked line 18 of the Dockerfile where it's clearly copying over rootfs. Which is an accompanying folder to the dockerfile. Knowing this I took a proper look at the permissions of rootfs/usr/bin/install-chrome.sh and found it missing it's executable bit. 

Problem resolved, not quite. 

I then received the following error:

/usr/bin/install-chrome.sh: line 5: TARGETPLATFORM: unbound variable
The command '/bin/sh -c apt-dpkg-wrap apt-get update &&     apt-dpkg-wrap apt-get install -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" jibri libgl1-mesa-dri procps jitsi-upload-integrations jq pulseaudio dbus dbus-x11 rtkit &&     /usr/bin/install-chrome.sh &&     apt-cleanup &&     adduser jibri rtkit' returned a non-zero code: 1

Taking a look at the Dockerfile line 11 ARG for TARGETPLATFORM is null and not strictly set, so I made it:

VAR TARGETPLATFORM=''

This cleared that error and I've now got this working.  One thing I did differently was to build the Dockerfile into an image with a tag that signifies I created it and then updated the compose file to point to that unique image name on the system. Less confusion for me later on. 

Zip installed using this method you suggested perfectly and I could also expand this as necessary, so thank you very much for the assistance.
